src/components/flat-table/flat-table.component.js

```diff
@@ -44,12 +44,15 @@ const FlatTable = ({
   return (
     <SidebarContext.Consumer>
       {(context) => (
-        <StyledFlatTableRoot {...filterStyledSystemMarginProps(rest)}>
+        <Box
+          {...filterStyledSystemMarginProps(rest)}
+          height={addDefaultHeight && !hasMaxHeight ? "100%" : height}
+          maxHeight={hasMaxHeight ? "100%" : undefined}
+        >
           <StyledFlatTableBox
             {...rest}
             {...((hasStickyHead || hasStickyFooter) && { overflowY: "auto" })}
-            height={addDefaultHeight && !hasMaxHeight ? "100%" : height}
-            maxHeight={hasMaxHeight ? "100%" : undefined}
+            height={footer ? "calc(100% - 40px)" : "100%"}
```

Comment:
Shouldn't this logic be part of the `StyledFlaTableBox`?

Reply:
I don't think so no, it has a height prop so we might as well use it